core/src/main/scala/com/commercetools/queue/MessageBatch.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2024 Commercetools GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.commercetools.queue
+
+import fs2.Chunk
+
+/**
+ * Interface to interact with the message received from a queue
+ * as a single batch allowing user to ack/nack all messages in a single
+ * call if the underlying implementation supports for it.
+ *
+ * For implementations that do not support batched acknowledging both
+ * `ackAll` and `nackAll` methods do not guarantee atomicity and will
+ * fallback to using per-message calls.
+ */
+trait MessageBatch[F[_], T] {
```

Comment:
To avoid problem in the future, can you seal this trait, and used the _unsealed_ approach we used for other entity? Basically it ensures that we can add new methods to the trait in the future without breaking binary compatibility for code using our library.

```suggestion
sealed trait MessageBatch[F[_], T] {
```

In the same file, below add:

```scala
private[queue] trait UnsealedMessageBatch[F[_], T] extends MessageBatch[F, T]
```

and our concrete implementations extends the _unsealed_ version instead.